src/Void.Core/BufferMessages.fs

```diff
@@ -0,0 +1,42 @@
+﻿namespace Void.Core
+
+type BufferMessage = inherit Message
+
+type BufferEnvelopeMessage<'TBufferMessage when 'TBufferMessage :> BufferMessage> =
+    {
+        BufferId : int
+        Message : 'TBufferMessage
+    }
+    interface EnvelopeMessage<'TBufferMessage>

```

Comment:
Yeah my idea is that the `BufferListService` may ultimately want to be able to handle buffer messages generically, just extract the buffer that is applies to, and delegate to a `BufferService` which knows how to handle that. There are going to be a lot of buffer-specific messages and I would like to separate the complexity of the buffer list from the actual meat of the buffer logic.

That said, it might be a premature abstraction.